### PR TITLE
[DOXIA-684] avoid flexmark-all: declare used extensions only

### PR DESCRIPTION
**Pull request by hboutemy**
https://issues.apache.org/jira/browse/DOXIA-684

### CODE REVIEW

**Comment by michael-o** [comment]
Nice, how did you determine that list?

**Comment by hboutemy** [comment]
just used https://github.com/vsch/flexmark-java/blob/master/flexmark-all/pom.xml to know the existing artifactIds, then added the missing ones for the used classes: nothing hard, just some not funny time to find the 8 ones :)

File: doxia-modules/doxia-module-markdown/pom.xml [hunk 1]
```diff
@@ -51,19 +51,57 @@ under the License.
     </contributor>
   </contributors>
 
+  <properties>
+    <flexmark.version>0.42.14</flexmark.version>
```
Comment (michael-o) [inline doxia-modules/doxia-module-markdown/pom.xml:55]: I think we should use the camelCase approach for version properties everywhere now: `flexmarkVersion`

Comment (michael-o) [reply doxia-modules/doxia-module-markdown/pom.xml:55]: master is at https://github.com/apache/maven-doxia/blob/master/doxia-modules/doxia-module-markdown/pom.xml#L67 `0.62.2`, 1.11.x requires a different version though

**Review by michael-o — CHANGES_REQUESTED** [review]

**Comment by michael-o** [comment]
Yet another issue:
```
[INFO] --- maven-dependency-plugin:3.1.1:analyze (default-cli) @ doxia-module-markdown ---
[WARNING] Used undeclared dependencies found:
[WARNING]    com.vladsch.flexmark:flexmark-util:jar:0.42.14:compile
```

File: doxia-modules/doxia-module-markdown/pom.xml (hunk 1 above)
Comment (hboutemy) [reply doxia-modules/doxia-module-markdown/pom.xml:55]: done the camelCase, you're right, it's what we do for parent POMs
on version, yes, the current PR is for Doxia 1.x, Doxia 2.x will get another version

Comment (michael-o) [reply doxia-modules/doxia-module-markdown/pom.xml:55]: If for 1.x then then PR is wrong. IT targets master.

**Comment by hboutemy** [comment]
merged in 539428bd17e649c506e4a7a549b8f8b5d1cc3c18 for Doxia 1.x and e6848f9e4b2fdeaa3378183d8b557d55fb476ad6 for Doxia 2

**Review by michael-o — COMMENTED** [review]
LGTM now

**Pull request closed without merging**